src/NzbDrone.Core/Configuration/ConfigFileProvider.cs

```diff
@@ -216,6 +223,36 @@ public string UrlBase
             }
         }
 
+        public UseProxy ProxyEnabled
+        {
+            get { return GetValueEnum<UseProxy>("ProxyEnabled", UseProxy.Off); }
+        }
+
+        public ProxyType ProxyType
+        {
+            get { return GetValueEnum<ProxyType>("ProxyType", ProxyType.Http); }
+        }
+
+        public string ProxyHostname
+        {
+            get { return GetValue("ProxyHostname", "localhost"); }
```

Comment:
It would just need a call to `ConfigService.SaveConfigDictionary), as we do for the ConfigFileService: https://github.com/Sonarr/Sonarr/blob/develop/src/NzbDrone.Api/Config/HostConfigModule.cs#L77

If the service doesn't handle that config value it just ignores it.

Probably best to move it to the ConfigService and keep as little as possible (just what we need during startup/to let a user disable auth) in the config file.